### PR TITLE
[No-JIRA] Run AndroidX and Appcompat migrations 

### PR DESCRIPTION
**Pull request by amolgupta**
Runs the `Android Studio` tools : 
- Migrate to AppCompat
- Migrate to AndroidX

### CODE REVIEW

File: Backpack/src/androidTest/java/net/skyscanner/backpack/dialog/BpkDialogTest.kt [hunk 1]
```diff
@@ -25,7 +25,7 @@ import org.junit.runner.RunWith
 class BpkDialogTest {
 
   private lateinit var handler: Handler
-  private lateinit var mActivity: Activity
+  private lateinit var mActivity: AppCompatActivity
```
Comment (antalindisguise) [inline Backpack/src/androidTest/java/net/skyscanner/backpack/dialog/BpkDialogTest.kt]: Can we maybe drop the `m` prefix?

File: Backpack/src/main/res/values/backpack.button.xml [hunk 2]
```diff
@@ -6,9 +6,9 @@
   </declare-styleable>
 
   <style name="bpkButtonBase">
-    <item name="android:fontFamily">sans-serif-light</item>
+    <item name="fontFamily">sans-serif-light</item>
```
Comment (antalindisguise) [inline Backpack/src/main/res/values/backpack.button.xml]: Are these changes intended?

File: app/build.gradle [hunk 3]
```diff
@@ -45,13 +45,13 @@ dependencies {
   implementation "androidx.appcompat:appcompat:$rootProject.ext.androidx"
   implementation "androidx.legacy:legacy-support-v4:$rootProject.ext.androidx"
   implementation "com.google.android.material:material:$rootProject.ext.androidx"
-  implementation 'com.android.support.constraint:constraint-layout:1.1.2'
+  implementation 'androidx.constraintlayout:constraintlayout:1.1.2'
```
Comment (antalindisguise) [inline app/build.gradle]: Could you bump it to `1.1.3` maybe as the latest stable?

File: Backpack/src/main/res/values/backpack.button.xml (hunk 2 above)
Comment (amolgupta) [reply Backpack/src/main/res/values/backpack.button.xml]: These are done by the migration tool.

File: Backpack/src/androidTest/java/net/skyscanner/backpack/dialog/BpkDialogTest.kt (hunk 1 above)
Comment (amolgupta) [reply Backpack/src/androidTest/java/net/skyscanner/backpack/dialog/BpkDialogTest.kt]: I can take this up as a separate PR and do it in the complete codebase.

File: Backpack/src/main/res/values/backpack.button.xml (hunk 2 above)
Comment (Bence1001) [reply Backpack/src/main/res/values/backpack.button.xml]: Removing the `android:` prefix means we are overriding `fontFamily` in the AppCompat Theme. If this is what we want then it's correct.
